### PR TITLE
feat(lists): make working with lists a breeze

### PR DESCRIPTION
**Pull request by johncorrelli**
when adding a list:
- hitting `enter` should start the next line with the same list character and indentation as the previous line
- hitting tab for a list line indents the line
- hitting shift tab for a list line outdents the line
- list lines and checklist lines are handled the same
- tests!
  - coverage for the new `keyboard-events`: 100%!
Screenshot from 2019-11-10 19-07-17

### CODE REVIEW

**Comment by johncorrelli** [comment]
paging @bradbumbalough :phone:

**Comment by bradbumbalough** [comment]
This looks awesome! I'm going to try and get into this more tonight.

File: yarn.lock [hunk 1]
```diff
@@ -50,151 +50,159 @@
     source-map "^0.5.0"
 
 "@babel/core@>=7.2.2", "@babel/core@^7.1.0", "@babel/core@^7.1.6", "@babel/core@^7.4.5":
-  version "7.6.4"
```
Comment (bradbumbalough) [inline yarn.lock]: Do you know what happened with yarn?

Comment (johncorrelli) [reply yarn.lock]: Yeah I kinda broke this and had to reinstall... i did yarn add jest, but that is already installed with create react app, so it caused mad hiccups. I had to torch node_modules and re yarn install. (that was the recommendation from yarn).

**Pull request merged**